BWIndex.py

```diff
@@ -0,0 +1,239 @@
+		self.SS[newNode] = suffixStart
+		self.nodeCount += 1
+
+	def splitNode(self, node, index):
+		# introduce a new node whose parent is the original parent of node, make it the new parent of node
+		newNode = self.nodeCount
+		parent = self.H[node]
+		self.G[newNode] = [node]
+		self.G[parent].remove(node)
+		self.G[parent] = self.G.get(parent, []) + [newNode]
+		self.H[newNode] = parent
+		self.H[node] = newNode
+		self.NS[newNode] = self.NS[node]
+		self.NE[newNode] = index - 1
+		self.NS[node] = index
+		self.nodeCount += 1
+		return newNode
+
+	def findMatchingChild(self, node, loc):
+		for child in self.G[node]:
+			if self.Text[loc] == self.Text[self.NS[child]]:
+				return child
+		return -1
+
+	def firstMismatch(self, node, ix):
+		offset = 1
+		while True:
+			if ix + offset >= len(self.Text):
+				return self.NS[node] + offset
+			if self.NS[node] + offset > self.NE[node]:
+				return -1
+			if self.Text[self.NS[node] + offset] != self.Text[ix + offset]:
+				return self.NS[node] + offset
+			offset += 1
+
+	def threadSuffix(self, ix):
+		'''
+		look for a node that matches the start of the current substring
+		if none found, create a new node and set its string to the current substring and SS to the initial index
+		if one is found, look for a mismatch within the string
+		if a mismatch is found, split the target node and append the remainder of the substring
+		if no mismatch, advance curIx by the length of the node string and iterate
+		'''
+		node = 0
+		curIx = ix
+		while True:
+			nextNode = self.findMatchingChild(node, curIx)
+			if nextNode == -1:
+				self.appendNode(node, curIx, ix)
+				break
+			node = nextNode
+			splitLoc = self.firstMismatch(node, curIx)
+			if splitLoc == -1:
+				curIx += self.NE[node] - self.NS[node] + 1
+				continue
+			newNode = self.splitNode(node, splitLoc)
+			self.appendNode(newNode, splitLoc + (curIx - self.NS[newNode]), ix)
+			c = self.nodeCount - 1
+			break
+
+	def computeCountArray(self):
+		for i in self.alphabet:
+			self.Count[i] = [0] * (len(self.Text) + 1)  # check does this cause any reference issues?
+		for j in range(1, len(self.BWT) + 1):
+			self.Count[self.BWT[j - 1]][j] = self.Count[self.BWT[j - 1]][j - 1] + 1
+			for k in self.alphabet:
+				if k != self.BWT[j - 1]:
+					self.Count[k][j] = self.Count[k][j - 1]
+		return (self.Count)
+
+	def FindCount(self, Pattern):
+		top = 0
+		bottom = len(self.BWT) - 1
+		while top <= bottom:
+			if len(Pattern) > 0:
+				symbol = Pattern[-1]
+				Pattern = Pattern[:-1]
+				if self.Count[symbol][bottom + 1] - self.Count[symbol][top] > 0:
+					top = self.FirstOccurrence[symbol] + self.Count[symbol][top]
+					bottom = self.FirstOccurrence[symbol] + self.Count[symbol][bottom + 1] - 1
+				else:
+					return 0
+			else:
+				return bottom - top + 1
+
+	def loadGenomeFromFile(self, genomefile):
+		lines = []
+		with open(genomefile) as f:
+			for line in f:
+				if line != '':
+					if line[0] != '>':

```

Comment:
You can link multiple conditions in an `if`: 
```
if line != '' and line[0] != '>':
```

The `and` will short-circuit - if the first condition isn't true, the second won't be evaluated.